test.sh

```diff
@@ -1,394 +1,539 @@
+		AZURE_STORAGE_ACCOUNT="${AZURE_DEV_STORAGE_ACCOUNT}"
+		AZURE_STORAGE_KEY="${AZURE_DEV_STORAGE_KEY}"
+	EOF
 fi
 
-cat >> $envfile <<EOF
-AWS_PROFILE=$aws_profile
-CLUSTER_URL=$CLUSTER_URL
-DCOS_ENTERPRISE=$enterprise
-DCOS_FILES_PATH=$dcos_files_path
-DCOS_LOGIN_PASSWORD=$DCOS_LOGIN_PASSWORD
-DCOS_LOGIN_USERNAME=$DCOS_LOGIN_USERNAME
-DCOS_SSH_USERNAME=$ssh_user
-FRAMEWORK=$framework
-PACKAGE_REGISTRY_ENABLED=$package_registry
-PYTEST_ARGS=$PYTEST_ARGS
-S3_BUCKET=$S3_BUCKET
-SECURITY=$security
-STUB_UNIVERSE_URL=$STUB_UNIVERSE_URL
+cat >> "${env_file}" <<-EOF
+	AWS_PROFILE=${aws_profile}
+	CLUSTER_URL=${CLUSTER_URL}
+	DCOS_ENTERPRISE=${enterprise}
+	DCOS_FILES_PATH=${dcos_files_path}
+	DCOS_LOGIN_PASSWORD=${DCOS_LOGIN_PASSWORD}
+	DCOS_LOGIN_USERNAME=${DCOS_LOGIN_USERNAME}
+	DCOS_SSH_USERNAME=${ssh_user}
+	FRAMEWORK=${framework}
+	PACKAGE_REGISTRY_ENABLED=$package_registry
+	PYTEST_ARGS=${PYTEST_ARGS}
+	PYTHONPATH=${WORK_DIR}/testing
+	S3_BUCKET=${S3_BUCKET}
+	SECURITY=${security}
+	STUB_UNIVERSE_URL=${STUB_UNIVERSE_URL}
 EOF
 
-while read line; do
-    # Prefix match, then strip prefix in envfile:
-    if [[ "${line:0:8}" = "TEST_SH_" ]]; then
-        echo ${line#TEST_SH_} >> $envfile
-    fi
+while read -r line; do
+  # Prefix match, then strip prefix in env_file.
+  if [[ "${line:0:8}" = "TEST_SH_" ]]; then
+    echo "${line#TEST_SH_}" >> "${env_file}"
+  fi
 done < <(env)
 
-if [ -n "$env_passthrough" ]; then
-    # If the -e flag is specified, add the ENVVAR lines for the
-    # comma-separated list of envvars
-    for envvar_name in ${env_passthrough//,/ }; do
-        echo "$envvar_name" >> $envfile
-    done
+if [ -n "${env_passthrough}" ]; then
+  # If the -e flag is specified, add the ENVVAR lines for the comma-separated
+  # list of envvars.
+  for envvar_name in ${env_passthrough//,/ }; do
+    echo "${envvar_name}" >> "${env_file}"
+  done
 fi
 
-if [ -n "$envfile_input" ]; then
-    cat "${envfile_input}" >> $envfile
+if [ -n "${env_file_input}" ]; then
+  cat "${env_file_input}" >> "${env_file}"
 fi
 
-CMD="docker run --rm \
--t \
-${docker_interactive_arg} \
---env-file $envfile \
-${volume_args} \
--w $WORK_DIR \
-${docker_image} \
-${docker_command}"
+################################################################################
+######################### Prepare and run command ##############################
+################################################################################
+
+docker pull "${docker_image}"
+
+set +x
+
+CMD="docker run
+    --rm
+    -t
+    ${docker_interactive_arg}
+    --env-file ${env_file}
+    -w ${WORK_DIR}
+    ${container_volumes}
+    ${docker_options}
+    ${docker_image}
+    ${docker_command}"
 
-echo "==="
+echo "================================================================================"
 echo "Docker command:"
-echo "  $CMD"
-echo ""
-echo "Environment:"
-while read line; do
-    echo "  $line"
-done <$envfile
-echo "==="
-
-$CMD
+# shellcheck disable=SC2001
```

Comment:
just to save others: 
```suggestion
# shellcheck disable=SC2001
# https://github.com/koalaman/shellcheck/wiki/SC2001
```
wasn't sure if putting it on the same line as the `disable=` would cause issue with shellcheck

Reply:
It would.